atest/testdata/running/for.robot

```diff
@@ -317,6 +317,13 @@ For In Range With Expressions Containing Floats
     \    @{result} =    Create List    @{result}    ${i}
     Should Be True    @{result} == [3.14, 1.14, -0.86]
 
+For In Zip
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:e', 'b:f', 'c:g', 'd:h']
+
```

Comment:
More tests needed. Some of these are likely to require enhancements to code too.

1. Handling lists with different number of elements. Probably should work like in Python here and just stop when the shortest list is empty?

2. Handling more than one list like `:FOR  ${a}  ${b}  ${c}  IN ZIP  ${x}  ${y}  ${z}`.

3. Handling non-lists. Should accept anything iterable except strings and fail with a clear error message if invalid data given. You can use `utils.is_list_like` to verify inputs.

4. Different number of variables than lists. Having just one variable works in Python (e.g. `for i in zip(x, y)`), but looking at the implementation it might be hard to support here. I'd be fine with a clear error if num(vars) != num(lists).